pywbemtools/pywbemcli/_cmd_listener.py

```diff
@@ -0,0 +1,473 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Click Command definition for the listener command group which includes
+cmds to manage indication listeners that are separate invocations of
+pywbemcli representing a listener.
+
+NOTE: Commands are ordered in help display by their order in this file.
+"""
+
+from __future__ import absolute_import, print_function
+
+import subprocess
+import signal
+import argparse
+from time import sleep
+import click
+import psutil
+
+from pywbem import WBEMListener
+
+from .pywbemcli import cli
+from ._common import CMD_OPTS_TXT, GENERAL_OPTS_TXT, SUBCMD_HELP_TXT, \
+    format_table
+from ._common_options import add_options, help_option
+from ._click_extensions import PywbemcliGroup, PywbemcliCommand
+
+DEFAULT_LISTENER_PORT = 5988
+DEFAULT_LISTENER_PROTOCOL = 'http'
+
+LISTEN_OPTIONS = [
+    click.option('--port', type=int, metavar='PORT',
+                 required=False, default=DEFAULT_LISTENER_PORT,
+                 help=u'The port number listened on. '
+                 'Default: {}'.format(DEFAULT_LISTENER_PORT)),
+    click.option('--protocol', type=click.Choice(['http', 'https']),
+                 metavar='PROTOCOL',
+                 required=False, default=DEFAULT_LISTENER_PROTOCOL,
+                 help=u'The protocol used by the listener (http, https). '
+                 'Default: {}'.format(DEFAULT_LISTENER_PROTOCOL)),
+    click.option('--certfile', type=str, metavar='FILE',
+                 required=False, default=None,
+                 help=u'Path name of a PEM file containing the certificate '
+                 'that will be presented as a server certificate during '
+                 'SSL/TLS handshake. Required when using https. '
+                 'The file may in addition contain the private key of the '
+                 'certificate.'),
+    click.option('--keyfile', type=str, metavar='FILE',
+                 required=False, default=None,
+                 help=u'Path name of a PEM file containing the private key '
+                 'of the server certificate. '
+                 'Required when using https and when the certificate file '
+                 'does not contain the private key.'),
+]
+
+
+class ListenerProperties(object):
+    """
+    The properties of a running named listener.
+    """
+
+    def __init__(self, pid, name, port, protocol, general_options):
+        self._pid = pid
+        self._name = name
+        self._port = port
+        self._protocol = protocol
+        self._general_options = general_options
+
+    def as_string_tuple(self):
+        """Return the object as a tuple of strings"""
+        return (str(self._pid), self._name, str(self._port), self._protocol,
+                ' '.join(self._general_options))
+
+    @property
+    def pid(self):
+        """int: Process ID of the listener"""
+        return self._pid
+
+    @property
```

Comment:
Suggest adding start time as property on listener.

Reply:
Good idea! Added to TODO items. DEFERRED to future update of this PR.